### PR TITLE
Correct typo of 'does' to 'do' in -map-when docs

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
Just noticed this when reading through.

### CODE REVIEW

**Comment by magnars** [comment]
Thanks! :+1:

File: dash.el [hunk 1]
```diff
@@ -357,7 +357,7 @@ See also: `-each-indexed'."
        (nreverse ,r))))
 
 (defun -map-when (pred rep list)
-  "Return a new list where the elements in LIST that does not match the PRED function
+  "Return a new list where the elements in LIST that do not match the PRED function
```
Comment (Fuco1) [inline dash.el]: We should probably either drop `where` or `that`, or change the `where` to `with`.  This way it makes no sense to me.  There should also be a period at the end.

Comment (GuiltyDolphin) [inline dash.el]: So perhaps:

```
Return a new list with the same elements as LIST, but with any elements satisfying PRED first mapped through REP.
```

?

Comment (Fuco1) [inline dash.el]: Actually, I confused this function with `-remove` :D The original is fine, sorry for the confusion.

(Though according to the emacs doc conventions, first line should be <80 char complete sentence... I don't see any way to describe this in such a short sentence :O)

Comment (GuiltyDolphin) [inline dash.el]: Yes, it can be a little restrictive - especially considering verbosity (for clarity) is often encouraged in naming and descriptions.

**Pull request merged**